### ISSUE TITLE
Add __len__ to ItemPaged class

### ISSUE THREAD

**Issue by akshaybabloo** [Azure/azure-sdk-for-python#9970]
**Is your feature request related to a problem? Please describe.**
For the class `ItemPaged` at

https://github.com/Azure/azure-sdk-for-python/blob/90fc2a709d9a20f7a3659255320c98d3beb8aa1e/sdk/core/azure-core/azure/core/paging.py#L84

It would be nice to add `__len__` function.

**Describe the solution you'd like**

Because it's an iterator using `len()` on it will give `TypeError: object of type 'ItemPaged' has no len()`. Yes you can use `len(list())`, but it's too wordy.

**Comment by kaerm** [comment]
@akshaybabloo thanks for the feedback! Someone from our team will take a look into this //cc: @lmazuel

**Comment by lmazuel** [comment]
@akshaybabloo the problem here is that we don't necessarly know in advance the length of this iterable. This is lazy loading, as you move around the iterator.
https://docs.python.org/3.8/library/collections.abc.html?highlight=abc

as you can see an iterator has no len

**Comment by xiangyan99** [comment]
Agree with Laurent. For paged item, we only know the length of this page which is not the real length of the operation. Close as by design but feel free to reopen it if you have concern.

**Comment by mcapuccini** [comment]
This is such a basic requirement, couldn't you add a count function? There are a lot of lazily loaded collections that do this.

**Comment by josiahdc** [comment]
ditto to the above, __len__ is such a fundamental attribute of any collection
